src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1205,17 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            try
```

Comment:
Well, in fact, this function does get hit. Maybe there is something bugs in the canSwitchToBroker(), I will debug it and figure it out.